sunpy/instr/aia.py

```diff
@@ -49,7 +50,15 @@ def aiaprep(aiamap):
         scale = 0.6*u.arcsec # pragma: no cover # can't test this because it needs a full res image
     scale_factor = aiamap.scale.x / scale
 
-    newmap = aiamap.rotate(recenter=True, scale=scale_factor.value, missing=aiamap.min())
+    tempmap = aiamap.rotate(recenter=True, scale=scale_factor.value, missing=aiamap.min())
+
+    # extract center from padded aiamap.rotate output
+    center = np.floor(tempmap.meta['crpix1'])
+    leftx = center - aiamap.data.shape[0] / 2
+    rightx = center + aiamap.data.shape[0] / 2
+    newmap = tempmap.submap([leftx, rightx]*u.pix, [leftx, rightx]*u.pix)
```

Comment:
I'm not sure why you are using just `shape[0]`, is it because the image need to be square?
In any case, another way of doing this without repetition would be:
```python
range_side = (center + np.array([-1, 1]) * aiamap.data.shape[0] / 2) * u.pix
newmap = tempmap.submap(range_side, range_side)
```